yugabyte/src/yugabyte/ysql/client.clj

```diff
@@ -57,6 +58,17 @@
        :close close-conn
        :log?  true})))
 
+(defprotocol YSQLYbClient
+  "Used by defclient macro in conjunction with jepsen.client/Client specifying actual logic"
+  (setup-cluster! [client test c conn-wrapper]
+    "Called once on a random node to set up database/cluster state for testing.")
+  (invoke-inner! [client test operation c conn-wrapper]
+    "Apply an operation to the client, returning an operation to be appended to the history.
+    This function is wrapped in with-errors, with-conn and with-retry")
```

Comment:
What is wrapped and how by `defclient` should be described in `defclient` comments, because wrapping it responsibility and function of `defclient`. `YSQLYbClient` is simply an interface.